lib/hexpm/accounts/web_auth.ex

```diff
@@ -0,0 +1,117 @@
+defmodule Hexpm.Accounts.WebAuth do
+  use Hexpm.Context
+
+  @moduledoc false
+
+  # A pool for storing and validating web auth requests.
+
+  alias Hexpm.Accounts.WebAuthRequest, as: Req
+  alias Hexpm.Accounts.Keys
+
+  # `device_code` refers to the code assigned to a client to identify it
+  # `user_code` refers to the code the user enters to authorize a client
+  # `scope` refers to the permissions granted to a key
+  # `scopes` refers to the list of scopes that are allowed in a web auth request
+  # `key_permission` is the default permissions given to generate a hex api key
+  # `key_params` is the default params to generate a hex api key
+
+  @key_permission %{domain: "api", resource: nil}
+  @key_params %{name: nil, permissions: [@key_permission]}
+
+  @doc """
+  Adds a web auth request to the pool and returns the response.
+
+  ## Params
+
+    - `scope` - Scope of the key to be generated. One of read and write.
+  """
+
+  def get_code(scope) do
+    device_code = "foo"
+    user_code = "bar"
+
+    request = %{
+      device_code: device_code,
+      user_code: user_code,
+      scope: scope,
+      verified: false
```

Comment:
we should not be passing internal fields like "verified" through changeset cast. cast is only for user input. Instead, make `verified: false` the default, and later on, do:

```elixir
changeset |> put_change(:verified, true)
```

Reply:
Fixed at https://github.com/hexpm/hexpm/pull/1058/commits/a83a2bf34a0389fd8b79d91b63ad093a62f97643. Have a look.